internal/resources/kubernetessecret/resource_secret.go

```diff
@@ -165,71 +254,214 @@ func resourceSecretDelete(_ context.Context, d *schema.ResourceData, m interface
+				}
+			}
+		case commonscope.UnknownScope:
+			return diag.Errorf("no valid scope type block found: minimum one valid scope type block is required among: %v. Please check the schema.", strings.Join(scope.ScopesAllowed[:], `, `))
 		}
+	}
 
-		getResp.Secret.Meta.Labels = meta.Labels
-		getResp.Secret.Meta.Description = meta.Description
+	if d.HasChange(ExportKey) {
+		err := createDeleteSecretExport(d.Get(ExportKey).(bool), scopedFullnameData, config, d)
+		if err != nil {
+			return diag.FromErr(err)
+		}
 	}
 
-	getResp.Secret.Spec = spec.ConstructSpec(d)
+	return dataSourceSecretRead(ctx, d, m)
+}
 
-	_, err = config.TMCConnection.SecretResourceService.SecretResourceServiceUpdate(
-		&secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretRequest{
-			Secret: getResp.Secret,
-		},
-	)
-	if err != nil {
-		return diag.FromErr(errors.Wrapf(err, "unable to update Tanzu Mission Control secret entry, name : %s", d.Get(scope.ClusterNameKey)))
+func updateCheckForSpec(d *schema.ResourceData, atomicSpec *clustersecretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretSpec, scope commonscope.Scope) bool {
+	if !(spec.HasSpecChanged(d)) {
+		username := d.Get(helper.GetFirstElementOf(spec.SpecKey, spec.DockerConfigjsonKey, spec.UsernameKey))
+		password := d.Get(helper.GetFirstElementOf(spec.SpecKey, spec.DockerConfigjsonKey, spec.PasswordKey))
+		url := d.Get(helper.GetFirstElementOf(spec.SpecKey, spec.DockerConfigjsonKey, spec.ImageRegistryURLKey))
+
+		secretSpecData, _ := spec.GetEncodedSpecData(url.(string), username.(string), password.(string))
+
+		atomicSpec.Data = map[string]strfmt.Base64{
+			spec.DockerconfigKey: secretSpecData,
+		}
+
+		return false
 	}
 
-	if d.HasChange(ExportKey) {
-		secretexportName := d.Get(NameKey).(string)
+	var secretSpec *clustersecretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretSpec
 
-		if d.Get(ExportKey).(bool) {
-			secretexportRequest := &secretexportmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretExportRequest{
-				SecretExport: &secretexportmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretExport{
-					FullName: constructFullnameSecetExport(d),
-					Meta:     common.ConstructMeta(d),
-				},
-			}
+	switch scope {
+	case commonscope.ClusterScope:
+		secretSpec = spec.ConstructSpecForClusterScope(d)
+	case commonscope.ClusterGroupScope:
+		clusterGroupScopeSpec := spec.ConstructSpecForClusterGroupScope(d)
+		secretSpec = clusterGroupScopeSpec.AtomicSpec
+	}
 
-			_, err = config.TMCConnection.SecretExportResourceService.SecretExportResourceServiceCreate(secretexportRequest)
+	atomicSpec.SecretType = secretSpec.SecretType
+	atomicSpec.Data = secretSpec.Data
 
-			if err != nil {
-				return diag.FromErr(errors.Wrapf(err, "unable to create Tanzu Mission Control secret export entry, name : %s", secretexportName))
+	log.Printf("[INFO] updating secret spec")
```

Comment:
Is this needed?

Reply:
these are for dubugging, that's why I've added them